Makefile

```diff
@@ -87,7 +87,7 @@ $(GOPATH)/bin/minishift$(IS_EXE): $(ADDON_ASSET_FILE) vendor ## Builds the binar
 vendor:
 	dep ensure -v
 
-$(ADDON_ASSET_FILE): $(GOPATH)/bin/go-bindata ## Compiles the built-in add-on into the binary using go-bindata
+assets $(ADDON_ASSET_FILE): $(GOPATH)/bin/go-bindata ## Compiles the built-in add-on into the binary using go-bindata
```

Comment:
-1, If you really want to add a asset target then do following.

```diff
diff --git a/Makefile b/Makefile
index 9b9fedb1..edb2e776 100644
--- a/Makefile
+++ b/Makefile
@@ -82,25 +82,26 @@ __check_defined = \
 # Start of the actual build targets
 
 .PHONY: $(GOPATH)/bin/minishift$(IS_EXE)
-$(GOPATH)/bin/minishift$(IS_EXE): $(ADDON_ASSET_FILE) vendor ## Builds the binary into $GOPATH/bin
+$(GOPATH)/bin/minishift$(IS_EXE): assets vendor ## Builds the binary into $GOPATH/bin
        go install -tags "$(BUILD_TAGS)" -pkgdir=$(ADDON_BINDATA_DIR) -ldflags="$(VERSION_VARIABLES)" ./cmd/minishift
 vendor:
        dep ensure -v
 
-$(ADDON_ASSET_FILE): $(GOPATH)/bin/go-bindata ## Compiles the built-in add-on into the binary using go-bindata
+.PHONY: assets
+assets: $(GOPATH)/bin/go-bindata ## Compiles the built-in add-on into the binary using go-bindata
        @mkdir -p $(ADDON_BINDATA_DIR)
        go-bindata $(GO_BINDATA_DEBUG) -prefix $(ADDON_ASSETS) -o $(ADDON_ASSET_FILE) -pkg bindata $(ADDON_ASSETS)/...
 
 $(BUILD_DIR)/$(GOOS)-$(GOARCH):
        mkdir -p $(BUILD_DIR)/$(GOOS)-$(GOARCH)
 
-$(BUILD_DIR)/darwin-amd64/minishift: $(ADDON_ASSET_FILE) vendor $(BUILD_DIR)/$(GOOS)-$(GOARCH) ## Cross compiles the darwin executable and places it in $(BUILD_DIR)/darwin-amd64/minishift
+$(BUILD_DIR)/darwin-amd64/minishift: assets vendor $(BUILD_DIR)/$(GOOS)-$(GOARCH) ## Cross compiles the darwin executable and places it in $(BUILD_DIR)/darwin-amd64/minishift
        CGO_ENABLED=0 GOARCH=amd64 GOOS=darwin go build -tags "$(BUILD_TAGS)" -pkgdir=$(ADDON_BINDATA_DIR) --installsuffix cgo -ldflags="$(LDFLAGS)" -o $(BUILD_DIR)/darwin-amd64/minishift ./cmd/minishift
 
-$(BUILD_DIR)/linux-amd64/minishift: $(ADDON_ASSET_FILE) vendor $(BUILD_DIR)/$(GOOS)-$(GOARCH) ## Cross compiles the linux executable and places it in $(BUILD_DIR)/linux-amd64/minishift
+$(BUILD_DIR)/linux-amd64/minishift: assets vendor $(BUILD_DIR)/$(GOOS)-$(GOARCH) ## Cross compiles the linux executable and places it in $(BUILD_DIR)/linux-amd64/minishift
        CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -tags "$(BUILD_TAGS)" -pkgdir=$(ADDON_BINDATA_DIR) --installsuffix cgo -ldflags="$(LDFLAGS)" -o $(BUILD_DIR)/linux-amd64/minishift ./cmd/minishift
 
-$(BUILD_DIR)/windows-amd64/minishift.exe: $(ADDON_ASSET_FILE) vendor $(BUILD_DIR)/$(GOOS)-$(GOARCH) ## Cross compiles the windows executable and places it in $(BUILD_DIR)/windows-amd64/minishift
+$(BUILD_DIR)/windows-amd64/minishift.exe: assets vendor $(BUILD_DIR)/$(GOOS)-$(GOARCH) ## Cross compiles the windows executable and places it in $(BUILD_DIR)/windows-amd64/minishift
        CGO_ENABLED=0 GOARCH=amd64 GOOS=windows go build -tags "$(BUILD_TAGS)" -pkgdir=$(ADDON_BINDATA_DIR) --installsuffix cgo -ldflags="$(LDFLAGS)" -o $(BUILD_DIR)/windows-amd64/minishift.exe ./cmd/minishift
 
 $(GOPATH)/bin/gh-release:
@@ -140,7 +141,7 @@ serve_docs: synopsis_docs ## Builds and serves the documentation using Middleman
 link_check_docs: gen_docs ## Checks the documentation for broken links
        cd docs && docker run -u $(DOCS_UID) $(DOC_VARIABLES) -tiv $(LOCAL_DOCS_DIR):$(CONTAINER_DOCS_DIR):Z $(DOCS_BUILDER_IMAGE) link_check
 
-$(DOCS_SYNOPISIS_DIR)/*.md: vendor $(ADDON_ASSET_FILE)
+$(DOCS_SYNOPISIS_DIR)/*.md: vendor assets
        @# https://github.com/golang/go/issues/15038#issuecomment-207631885 ( CGO_ENABLED=0 )
        DOCS_SYNOPISIS_DIR=$(DOCS_SYNOPISIS_DIR) CGO_ENABLED=0 go run -tags "$(BUILD_TAGS) gendocs" -ldflags="$(LDFLAGS)" gen_help_text.go
 
@@ -196,7 +197,7 @@ clean:
        rm -f  $(DOCS_SYNOPISIS_DIR)/*.md
 
 .PHONY: test
-test: vendor $(ADDON_ASSET_FILE)  ## Run unit tests
+test: vendor assets  ## Run unit tests
        @go test -v -tags "$(BUILD_TAGS)" -ldflags="$(VERSION_VARIABLES)" $(shell $(PACKAGES))
```

Reply:
Like if i do `make` without adding `assets` as a dependency but `ADDON_ASSET_FILE` it won't run go-bindata again and again but only  when changes are there, but if `assets` is added as dependency it will run `go-bindata` agian and again even with no changes.

Reply:
Please let me know if the above comments are not clear...

Reply:
@anjannath I am still not sure why you want this https://github.com/minishift/minishift/issues/2281#issuecomment-382269349 ?

Reply:
@LalatenduMohanty Replied in https://github.com/minishift/minishift/issues/2281#issuecomment-382273427

Reply:
@anjannath I am still not sure what workflow you are looking for. Is it ability to do do `make assets` and then just `make` without need of doing `make clean`?

Reply:
Also using a phony target will be better IMO, so that it gets executed every-time you do make

Reply:
> Is it ability to do do make assets and then just make without need of doing make clean?

Yes, partly because make after make clean takes some time for me so i was doing this locally when working on some add-on... and also it makes clear that we are building the assets and not cleaning the workspace.

Reply:
> Also using a phony target will be better IMO, so that it gets executed every-time you do make

If all the `ADDON_ASSET_FILE` references are replaced with `assets` as shown in https://github.com/minishift/minishift/pull/2282#discussion_r181663534 it gets executed every time we do `make`. Is that what you want?